File: Itido-BMS/RentAPI/Rent/Models/TimePlanning/Noti.cs
using System;
using Newtonsoft.Json;

namespace Rent.Models.TimePlanning
{
    public class Noti
    {
        public int ID { get; set; }
        public string NotiType { get; set; }
        public int SubjectID { get; set; }
        public int UserID { get; set; }

        [JsonConverter(typeof(DartDateTimeConverter))]
        [JsonProperty("sendTime")]
        public DateTime SendTime { get; set; }
        public int? SenderID { get; set; }
        public virtual User Sender { get; set; }

        public int ReceiverID { get; set; }
        public virtual User Receiver { get; set; }

        public string Title { get; set; }
        public string Body { get; set; }
        public bool Seen { get; set; }

        public static string AccidentReport = "AccidentReport_";
        public static string Absence = "Absence_";
        public static string Shift = "Shift_";

        public static string Work = "Work_";
        public static string Post = "Post_";
        public static string WorkContract = "WorkContract_";

        public static string WorkRegistration = "WorkRegistration";
        public static string Invitation = "Invitation";
        public static string AbsenceRequested = Absence + "Requested";
        public static string AbsenceCreated = Absence + "Created";
        public static string AbsenceApproved = Absence + "Approved";
        public static string AbsenceDeclined = Absence + "Declined";
        public static string ShiftInvitationReceived = Shift + "InvitationReceived";
        public static string ShiftInvitationApproved = Shift + "InvitationApproved";
        public static string ShiftInvitationDeclined = Shift + "InvitationDeclined";

        public static string AccidentReportRequested = AccidentReport + "Requested";
        public static string AccidentReportApproved = AccidentReport + "Approved";
        public static string AccidentReportDeclined = AccidentReport + "Declined";
        public static string WorkContractUserAdded = WorkContract + "UserAdded";
        public static string WorkContractUserRemoved = WorkContract + "UserRemoved";

        public static string WorkRegistrationApproved = WorkRegistration + "Approved";
        public static string WorkRegistrationDeclined = WorkRegistration + "Declined";

        public static string WorkReplacerAdded = Work + "ReplacerAdded";

        public static string InvitationSend = Invitation + "Send";
    }
}

File: Itido-BMS/dart/bms_dart/lib/src/models/noti.dart
/*
public int ID { get; set; }
  public string NotiType { get; set; }
  public int SubjectID { get; set; }

  public DateTime SendTime { get; set; }
  public int? SenderID { get; set; }
  public virtual User Sender { get; set; }

  public int ReceiverID { get; set; }
  public virtual User Receiver { get; set; }
  public bool Seen { get; set; }
*/

class Noti {
  int id;
  String title;
  String body;
  String senderName;
  DateTime sendTime;
  int subjectId;
  int userId;
  String notiType;
  bool seen;

  Noti({
    this.id,
    this.title,
    this.body,
    this.senderName,
    this.sendTime,
    this.subjectId,
    this.userId,
    this.notiType,
    this.seen,
  });
  Noti.fromJson(json)
      : this.id = json['id'],
        this.title = json['title'],
        this.body = json['body'],
        this.senderName = json['senderName'],
        this.sendTime = DateTime.parse(json['sendTime']),
        this.subjectId = json['subjectID'],
        this.userId = json['userID'],
        this.notiType = json['notiType'],
        this.seen = json['seen'];

  Map<String, dynamic> toMap() => {
        'id': this.id,
        'title': this.title,
        'body': this.body,
        'senderName': this.senderName,
        'sendTime': this.sendTime.toString(),
        'subjectId': this.subjectId,
        'userId': this.userId,
        'notiType': this.notiType,
        'seen': this.seen,
      };
}
